### PR TITLE
Removed alias fetchcontracts.domain.TemplateId

### PR DESCRIPTION
**Pull request by chunlokling-da**
fixes https://github.com/digital-asset/daml/issues/15040

### CODE REVIEW

**Comment by S11001001** [comment]
Since the existing `domain.TemplateId`s serve as an informal TODO for #15098, it might be better not to apply this expansion unconditionally until that issue is fixed.

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala [hunk 1]
```diff
@@ -290,7 +290,7 @@ object PackageService {
     ) => Error \/ (Option[ContractTypeId.Interface.Resolved], typesig.Type)
 
   type ResolveKeyType =
-    TemplateId.RequiredPkg => Error \/ typesig.Type
+    ContractTypeId.RequiredPkg => Error \/ typesig.Type
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala:293]:
```suggestion
    ContractTypeId.RequiredPkg => Error \/ typesig.Type // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala [hunk 2]
```diff
@@ -319,7 +319,7 @@ object PackageService {
     Map[Choice, NonEmpty[Map[Option[ContractTypeId.Interface.Resolved], typesig.Type]]]
   ]]
 
-  type KeyTypeMap = Map[TemplateId.RequiredPkg, typesig.Type]
+  type KeyTypeMap = Map[ContractTypeId.RequiredPkg, typesig.Type]
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala:322]:
```suggestion
  type KeyTypeMap = Map[ContractTypeId.RequiredPkg, typesig.Type] // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala [hunk 3]
```diff
@@ -385,7 +385,7 @@ object PackageService {
 
   def resolveKey(
       keyTypeMap: KeyTypeMap
-  )(templateId: TemplateId.RequiredPkg): Error \/ typesig.Type =
+  )(templateId: ContractTypeId.RequiredPkg): Error \/ typesig.Type =
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/PackageService.scala:388]:
```suggestion
  )(templateId: ContractTypeId.RequiredPkg): Error \/ typesig.Type = // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/CreateAndExercise.scala [hunk 4]
```diff
@@ -48,7 +48,7 @@ private[http] final class CreateAndExercise(
           decoder
             .decodeCreateCommand(reqBody, jwt, toLedgerId(jwtPayload.ledgerId))
             .liftErr(InvalidUserInput): ET[
-            domain.CreateCommand[ApiRecord, TemplateId.RequiredPkg]
+            domain.CreateCommand[ApiRecord, ContractTypeId.RequiredPkg]
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/endpoints/CreateAndExercise.scala:51]:
```suggestion
            domain.CreateCommand[ApiRecord, ContractTypeId.RequiredPkg] // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala [hunk 5]
```diff
@@ -39,15 +39,15 @@ class DomainJsonDecoder(
   type ET[A] = EitherT[Future, JsonError, A]
 
   def decodeCreateCommand(a: JsValue, jwt: Jwt, ledgerId: LedgerApiDomain.LedgerId)(implicit
-      ev1: JsonReader[domain.CreateCommand[JsValue, TemplateId.OptionalPkg]],
+      ev1: JsonReader[domain.CreateCommand[JsValue, ContractTypeId.OptionalPkg]],
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala:42]:
```suggestion
      ev1: JsonReader[domain.CreateCommand[JsValue, ContractTypeId.OptionalPkg]], // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala [hunk 6]
```diff
@@ -39,15 +39,15 @@ class DomainJsonDecoder(
...
       ec: ExecutionContext,
       lc: LoggingContextOf[InstanceUUID],
-  ): ET[domain.CreateCommand[lav1.value.Record, TemplateId.RequiredPkg]] = {
+  ): ET[domain.CreateCommand[lav1.value.Record, ContractTypeId.RequiredPkg]] = {
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala:45]:
```suggestion
  ): ET[domain.CreateCommand[lav1.value.Record, ContractTypeId.RequiredPkg]] = { // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala [hunk 7]
```diff
@@ -39,15 +39,15 @@ class DomainJsonDecoder(
...
     val err = "DomainJsonDecoder_decodeCreateCommand"
     for {
       fj <- either(
         SprayJson
-          .decode[domain.CreateCommand[JsValue, TemplateId.OptionalPkg]](a)
+          .decode[domain.CreateCommand[JsValue, ContractTypeId.OptionalPkg]](a)
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala:50]:
```suggestion
          .decode[domain.CreateCommand[JsValue, ContractTypeId.OptionalPkg]](a) // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala [hunk 8]
```diff
@@ -220,10 +220,10 @@ class DomainJsonDecoder(
         .flatMap(jsObj => jsValueToApiValue(lfType, jsObj).flatMap(mustBeApiRecord))
         .valueOr(e => spray.json.deserializationError(e.shows))
 
-  def templateRecordType(id: domain.TemplateId.RequiredPkg): JsonError \/ domain.LfType =
+  def templateRecordType(id: domain.ContractTypeId.RequiredPkg): JsonError \/ domain.LfType =
     resolveTemplateRecordType(id).liftErr(JsonError)
 
-  def keyType(id: domain.TemplateId.OptionalPkg)(implicit
+  def keyType(id: domain.ContractTypeId.OptionalPkg)(implicit
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala:226]:
```suggestion
  def keyType(id: domain.ContractTypeId.OptionalPkg)(implicit // TODO #15098 .Template
```

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala [hunk 9]
```diff
@@ -392,9 +392,10 @@ object JsonProtocol extends JsonProtocolLow {
   )
 
   implicit val CreateCommandFormat
-      : RootJsonFormat[domain.CreateCommand[JsValue, domain.TemplateId.OptionalPkg]] = jsonFormat3(
-    domain.CreateCommand[JsValue, domain.TemplateId.OptionalPkg]
-  )
+      : RootJsonFormat[domain.CreateCommand[JsValue, domain.ContractTypeId.OptionalPkg]] =
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala:395]:
```suggestion
      : RootJsonFormat[domain.CreateCommand[JsValue, domain.ContractTypeId.OptionalPkg]] = // TODO #15098 .Template
```

**Review by S11001001 — COMMENTED** [review]
Some of these `.Template` should be trivial and therefore doable in this PR. Some of them will not be trivial, in which case the TODO comment suggested can be left.

**Review by ray-roestenburg-da — COMMENTED** [review]
As @stephen said :-) please add TODO's for Template cases.

**Comment by chunlokling-da** [comment]
this issue depends on https://github.com/digital-asset/daml/issues/15098

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala [hunk 10]
```diff
@@ -220,10 +224,11 @@ class DomainJsonDecoder(
         .flatMap(jsObj => jsValueToApiValue(lfType, jsObj).flatMap(mustBeApiRecord))
         .valueOr(e => spray.json.deserializationError(e.shows))
 
-  def templateRecordType(id: domain.TemplateId.RequiredPkg): JsonError \/ domain.LfType =
+  def templateRecordType(id: domain.ContractTypeId.RequiredPkg): JsonError \/ domain.LfType =
     resolveTemplateRecordType(id).liftErr(JsonError)
 
-  def keyType(id: domain.TemplateId.OptionalPkg)(implicit
+  def keyType(id: domain.ContractTypeId.OptionalPkg)(
+      implicit // TODO #15098 .Template
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/DomainJsonDecoder.scala:231]: The comment might have gotten reformatted somewhere nonsensical here 🙂

File: ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala [hunk 11]
```diff
@@ -391,10 +391,12 @@ object JsonProtocol extends JsonProtocolLow {
     domain.CommandMeta
   )
 
-  implicit val CreateCommandFormat
-      : RootJsonFormat[domain.CreateCommand[JsValue, domain.TemplateId.OptionalPkg]] = jsonFormat3(
-    domain.CreateCommand[JsValue, domain.TemplateId.OptionalPkg]
-  )
+  implicit val CreateCommandFormat: RootJsonFormat[
+    domain.CreateCommand[JsValue, domain.ContractTypeId.OptionalPkg]
+  ] = // TODO #15098 .Template
```
Comment (S11001001) [inline ledger-service/http-json/src/main/scala/com/digitalasset/http/json/JsonProtocol.scala:396]: Likewise.

**Review by S11001001 — APPROVED** [review]

**Pull request merged**